extensions/grpc/codegen/src/main/java/io/quarkus/grpc/deployment/GrpcCodeGen.java

```diff
@@ -100,11 +101,14 @@ public boolean trigger(CodeGenContext context) throws CodeGenException {
             Collection<Path> protoFilesFromDependencies = gatherProtosFromDependencies(dirWithProtosFromDependencies, protoDirs,
                     context);
             if (!protoFilesFromDependencies.isEmpty()) {
-                protoFilesFromDependencies.stream()
-                        .map(Path::normalize)
-                        .map(Path::toAbsolutePath)
-                        .map(Path::toString)
-                        .forEach(protoFiles::add);
+                for (Path files : protoFilesFromDependencies) {
+                    var pathToProtoFile = files.normalize().toAbsolutePath();
+                    var pathToParentDir = files.getParent();
+                    // Add the proto file to the list of proto to compile, but also add the directory containing the
+                    // proto file to the list of directories to include (it's a set, so no duplicate).
```

Comment:
Why also the dir?

Reply:
Because if you don't, it cannot resolve import statements from proto files included in the application. That was the bug.